### PR TITLE
Upgrade to k8 1.7.0

### PR DESCRIPTION
**Pull request by lewismarshall**

### CODE REVIEW

**Comment by lewismarshall** [comment]
I should probably squash this up a bit and maybe wait until the @vaijab's label work is on keto-master but, this works:
```
/ # kubectl get nodes
NAME                                          STATUS                     AGE       VERSION
ip-172-31-17-97.eu-west-2.compute.internal    Ready,SchedulingDisabled   4m        v1.7.0+coreos.0
ip-172-31-26-40.eu-west-2.compute.internal    Ready,SchedulingDisabled   2m        v1.7.0+coreos.0
ip-172-31-27-166.eu-west-2.compute.internal   Ready                      1m        v1.7.0+coreos.0
ip-172-31-30-108.eu-west-2.compute.internal   Ready,SchedulingDisabled   2m        v1.7.0+coreos.0
/ # kubectl get pods
No resources found.
/ # kubectl get pods --namespace=kube-system
NAME                                                                  READY     STATUS    RESTARTS   AGE
canal-h7699                                                           3/3       Running   0          1m
canal-k53dh                                                           3/3       Running   0          2m
canal-p85mh                                                           3/3       Running   0          2m
canal-rw6fp                                                           3/3       Running   0          4m
keto-tokens-2k3kt                                                     1/1       Running   0          2m
keto-tokens-5qkt3                                                     1/1       Running   0          2m
keto-tokens-nsf84                                                     1/1       Running   0          4m
kube-apiserver-ip-172-31-17-97.eu-west-2.compute.internal             1/1       Running   0          1m
kube-apiserver-ip-172-31-26-40.eu-west-2.compute.internal             1/1       Running   0          1m
kube-apiserver-ip-172-31-30-108.eu-west-2.compute.internal            1/1       Running   0          1m
kube-controller-manager-ip-172-31-17-97.eu-west-2.compute.internal    1/1       Running   0          1m
kube-controller-manager-ip-172-31-26-40.eu-west-2.compute.internal    1/1       Running   0          1m
kube-controller-manager-ip-172-31-30-108.eu-west-2.compute.internal   1/1       Running   0          1m
kube-dns-2425271678-twvqj                                             2/3       Running   0          4m
kube-proxy-79z5q                                                      1/1       Running   0          2m
kube-proxy-bg7xj                                                      1/1       Running   0          1m
kube-proxy-trjtq                                                      1/1       Running   0          4m
kube-proxy-tvdfc                                                      1/1       Running   0          2m
kube-scheduler-ip-172-31-17-97.eu-west-2.compute.internal             1/1       Running   0          1m
kube-scheduler-ip-172-31-26-40.eu-west-2.compute.internal             1/1       Running   0          1m
kube-scheduler-ip-172-31-30-108.eu-west-2.compute.internal            1/1       Running   0          2m
```

File: pkg/kubeadm/kubeadm.go [hunk 1]
```diff
@@ -21,14 +21,15 @@ import (
 	"github.com/UKHomeOffice/keto-k8/pkg/constants"
 	"github.com/UKHomeOffice/keto-k8/pkg/etcd"
 	"github.com/UKHomeOffice/keto-k8/pkg/kubeadm/pkiutil"
+	"os"
```
Comment (gambol99) [inline pkg/kubeadm/kubeadm.go]: can move the "os" back to the standard lib group

Comment (lewismarshall) [inline pkg/kubeadm/kubeadm.go]: Indeed - un-intelij auto packaging ting!

**Comment by lewismarshall** [comment]
@gambol99 - good catch - intellij told off suitably and also squashed the pr a bit...

File: vendor/k8s.io/kubernetes/cmd/kubeadm/app/master/manifests.go [hunk 2]
```diff
@@ -369,36 +340,43 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool) [
 
 	defaultArguments := map[string]string{
 		"insecure-port":                     "0",
-		"admission-control":                 kubeadmconstants.DefaultAdmissionControl,
+		"admission-control":                 defaultv16AdmissionControl,
```
Comment (jaykeshur) [inline vendor/k8s.io/kubernetes/cmd/kubeadm/app/master/manifests.go]: Not defaultv17AdmissionControl ?

File: pkg/kubeadm/kubeadm.go [hunk 3]
```diff
@@ -162,6 +163,11 @@ func (k *Config) CreatePKI() (err error) {
 
 // CreateKubeConfig - Creates all the kubeconfig files requires for masters
 func (k *Config) CreateKubeConfig() (err error) {
+	if k.KubeletID == "" {
+		if k.KubeletID, err = os.Hostname(); err != nil {
```
Comment (vaijab) [inline pkg/kubeadm/kubeadm.go]: What would be the hostname? If kubelet runs in a container, does that mean that the hostname changes everytime kubelet restarts?

File: pkg/kubeadm/kubeadm.go [hunk 4]
```diff
@@ -28,7 +29,7 @@ import (
 const cmdKubeadm string = "kubeadm"
 
 var (
-	cmdOptsCerts      = []string{"alpha", "phase", "certs", "selfsign", "--cert-altnames"}
+	cmdOptsCerts      = []string{"alpha", "phase", "certs", "selfsign", "--apiserver-advertise-address", "0.0.0.0", "--cert-altnames"}
```
Comment (vaijab) [inline pkg/kubeadm/kubeadm.go]: Is it OK to have `0.0.0.0` as advertised address?

File: pkg/kubeadm/kubeadm.go (hunk 3 above)
Comment (lewismarshall) [inline pkg/kubeadm/kubeadm.go]: The container (keto-k8) runs with --net=host to properly support kubeadm for this and a hoot of other things (validation checks, systemd intergration etc)...

File: pkg/kubeadm/kubeadm.go (hunk 4 above)
Comment (lewismarshall) [inline pkg/kubeadm/kubeadm.go]: See https://kubernetes.io/docs/admin/kubeadm/ and search for `--apiserver-advertise-address`:

> If not set (or set to 0.0.0.0) then IP for the default interface will be used.

Unfortunately it is a required flag when generating certs - it wasn't previously (the integration tests found this :))

**Pull request merged**